clients/ordering/src/components/reducers/cartReducer.tsx

```diff
@@ -0,0 +1,91 @@
+import menucards from '../menucards';
+import {
+  ADD_TO_CART,
+  ADD_QUANTITY,
+  SUB_QUANTITY,
+  REMOVE_ITEM,
+  ADD_TIP,
+} from '../actions/action-types/cart-actions';
+
+const initState = {
+  items: menucards,
+  addedItems: [],
+  total: 0,
+};
+
+const cartReducer = (state: any = initState, action: any) => {
+  if (action.type === ADD_TO_CART) {
+    const addedItem = state.addedItems.find(
+      (item: { id: string }) => item.id === action.id
+    );
+    const existedItem = state.items.find(
+      (item: { id: string }) => item.id === action.id
+    );
+    if (addedItem) {
+      addedItem.quantity += 1;
```

Comment:
I would not mutate the quantity of the `addedItem`. I would suggest instead of using `find`, using `findIndex` to find where the existing item is in the array, and then create a new item with the `quantity` field updated and create a new array, so something like:
```
 const addedItemIndex = state.addedItems.findIndex(
      (item: { id: string }) => item.id === action.id
    );
if (addedItem > 0) {
  const newItems = [...state.addedItems]
  newItems[addedItemIndex].quantity += 1;
  return {
    ...state,
    addedItems: newItems
  }
}
```
This example copies the existing `addedItems` and mutates the copy instead of mutating the original item